libraries/SPI/SPI.cpp

```diff
@@ -307,27 +307,32 @@ uint8_t SPIClass::transfer(uint8_t data) {
 }
 
 uint16_t SPIClass::transfer16(uint16_t data) {
-    union {
-            uint16_t val;
-            struct {
-                    uint8_t lsb;
-                    uint8_t msb;
-            };
-    } in, out;
-    in.val = data;
-
-    if((SPI1C & (SPICWBO | SPICRBO))) {
-        //LSBFIRST
-        out.lsb = transfer(in.lsb);
-        out.msb = transfer(in.msb);
-    } else {
-        //MSBFIRST
-        out.msb = transfer(in.msb);
-        out.lsb = transfer(in.lsb);
-    }
-    return out.val;
+    while (SPI1CMD & SPIBUSY) {}
+    // Set to 16Bits transfer
+    setDataBits(16);
```

Comment:
Disclaimer: not an SPI expert or even a user. 

This change is forcing 16bit transfer. I think the "16" in transfer16() refers to the data type taken as arg, i. e. uint16_t, and not the transfer bit length. With this change it is no longer possible to send a 16 bit value as 2 8bit values with configured byte ordering, unless the user does 2 method calls. 
If my previous comments are correct, this is essentially changing the method semantics.
Do all SPI devices support 16bit transfers in this new way? 
Does it make sense to switch back and forth between 16 and 8 bit length modes for one particular device?
Does the current code not make sense? 
Is anyone out there using the current transfer16()?

Reply:
@devyte we already have a discussion regarding this problem here: https://github.com/esp8266/Arduino/issues/6417#issuecomment-522224896
And as @UlliBien explained below.